core/transaction.go

```diff
@@ -63,6 +63,16 @@ type Transaction interface {
 	Signature() []*felt.Felt
 }
 
+type AddInvokeTxResult struct {
+	Hash string `json:"transaction_hash"`
+	Code string `json:"code"`
+}
+type AddInvokeTxError struct {
+	Data    any    `json:"data"`
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
+
```

Comment:
These dont belong here, they should either be moved to `gateway` or `rpc`, which ever makes more sense

Reply:
Ive moved these to gateway since moving them to rpc causes an import cycle error. I've also replaced `AddInvokeTxError` with a new struct `StarknetError` so that it matches what both the sequencer and pathfinder returns